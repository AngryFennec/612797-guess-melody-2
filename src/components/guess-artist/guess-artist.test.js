import React from 'react';
import GuessArtist from './guess-artist';
import renderer from 'react-test-renderer';

const question = {
  type: `artist`,
  song: {
    artist: `Jim Beam`,
    src: `https://upload.wikimedia.org/wikipedia/commons/a/a3/Kimi_ga_Yo_instrumental.ogg`,
  },
  answers: [
    {
      picture: `http://placehold.it/134x134`,
      artist: `Artist 1`
    },
    {
      picture: `http://placehold.it/134x134`,
      artist: `Artist 2`
    },
    {
      picture: `http://placehold.it/134x134`,
      artist: `Artist 3`
    },
    {
      picture: `http://placehold.it/134x134`,
      artist: `Artist 4`
    },
  ],
};

it(`GuessArtist component renders correctly`, () => {
  const tree = renderer
    .create(<GuessArtist
      question = {question}
      onAnswer = {jest.fn()}
    />,
    {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
