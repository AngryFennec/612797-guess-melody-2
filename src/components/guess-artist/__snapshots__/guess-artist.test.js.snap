// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuessArtist component renders correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <button
      className="track__button track__button--play"
      disabled={true}
      onClick={[Function]}
      type="button"
    />
    <div
      className="track__status"
    >
      <audio />
    </div>
  </div>
  <form
    className="game__artist"
    onChange={[MockFunction]}
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-0"
        name="answer"
        type="radio"
        value="artist-0"
      />
      <label
        className="artist__name"
        htmlFor="artist-0"
      >
        <img
          alt="Artist 1"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Artist 1
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-1"
        name="answer"
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="artist-1"
      >
        <img
          alt="Artist 2"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Artist 2
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-2"
        name="answer"
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="artist-2"
      >
        <img
          alt="Artist 3"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Artist 3
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-3"
        name="answer"
        type="radio"
        value="artist-3"
      />
      <label
        className="artist__name"
        htmlFor="artist-3"
      >
        <img
          alt="Artist 4"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Artist 4
      </label>
    </div>
  </form>
</section>
`;
